src/doc/unstable-book/src/compiler-flags/source-based-code-coverage.md

```diff
@@ -0,0 +1,175 @@
+
+When running a coverage-instrumented program, the counter values are written to a `profraw` file at program termination. LLVM bundles tools that read the counter results, combine those results with the coverage map (embedded in the program binary), and generate coverage reports in multiple formats.
+
+## Enable coverage profiling in the Rust compiler
+
+*IMPORTANT:* Rust's coverage profiling features may not be enabled, by default. To enable them, you may need to build a version of the Rust compiler with the `profiler` feature enabled.
+
+First, edit the `config.toml` file, and find the `profiler` feature entry. It may be commented out:
+
+```toml
+# Build the profiler runtime (required when compiling with options that depend
+# on this runtime, such as `-C profile-generate` or `-Z instrument-coverage`).
+#profiler = false
+```
+
+Uncomment the feature entry and set it to `true`:
+
+```toml
+profiler = true
+```
+
+Then rebuild the Rust compiler (see [rustc-dev-guide-how-to-build-and-run]).
+
+### Building the demangler
+
+LLVM coverage reporting tools generate results that can include function names and other symbol references, and the raw coverage results report symbols using the compiler's "mangled" version of the symbol names, which can be difficult to interpret. To work around this issue, LLVM coverage tools also support a user-specified symbol name demangler. Rust's symbol name demangler can be built with:
+
+```shell
+$ ./x.py build rust-demangler
+```
+
+## Compiling with coverage enabled
+
+Set the `-Zinstrument-coverage` compiler flag in order to enable LLVM source-based code coverage profiling.
+
+With `cargo`, you can instrument your program binary *and* dependencies at the same time.
+
+For example (if your project's Cargo.toml builds a binary by default):
+
+```shell
+$ cd your-project
+$ cargo clean
+$ RUSTFLAGS="-Zinstrument-coverage" cargo build
+```
+
+If `cargo` is not configured to use your `profiler`-enabled version of `rustc`, set the path explicitly via the `RUSTC` environment variable. Here is another example, using a `stage1` build of `rustc` to compile an `example` binary (from the [`json5format`](https://crates.io/crates/json5format) crate):
+
+```shell
+$ RUSTC=$HOME/rust/build/x86_64-unknown-linux-gnu/stage1/bin/rustc \
+    RUSTFLAGS="-Zinstrument-coverage" \
+    cargo build --example formatjson5
+```
+
+## Running the instrumented binary to generate raw coverage profiling data
+
+In the previous example, `cargo` generated the coverage-instrumented binary `formatjson5`:
+
+```shell
+$ echo "{some: 'thing'}" | target/debug/examples/formatjson5 -
+```
+```json5
+{
+    some: 'thing',
+}
+```
+
+After running this program, a new file, `default.profraw`, should be in the current working directory:
+
+```shell
+$ ls -l default.profraw
+```
+```text
+-rw-r--r-- 1 user group 3.3M Oct 31 12:31 default.profraw
+```
```

Comment:
I'd remove the `ls` example here. Again, to verbose for my taste.

Reply:
Done